src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -512,7 +512,10 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 break;
         }
 
-        assert(ins != INS_invalid);
+        if (!(intrin.id == NI_AdvSimd_StoreSelectedScalar || intrin.id == NI_AdvSimd_Arm64_StoreSelectedScalar))
```

Comment:
The logic of detecting instruction should be moved in the above switch-case so when we leave, this assert holds true. Then below, you can just call `emitIns()`.

Reply:
Done. Also moved updating op2Reg to avoid duplication. Not sure if it's ideal though.